File: Design-Patterns-Using-Swift/Command/CommandPatternStock.swift
// Agent (Invoker)
class Agent {
    func placeOrder(command: Order) {
        command.execute()
    }
}

// Order (Command)
protocol Order {
    func execute()
}

// BuyStockOrder (ConcreteCommand)
class BuyStockOrder: Order {
    private let stock: ReceiverStockTrade

    init(stock: ReceiverStockTrade) {
        self.stock = stock
    }

    func execute() {
        stock.buy()
    }
}

// SellStockOrder (ConcreteCommand)
class SellStockOrder: Order {
    private let stock: ReceiverStockTrade

    init(stock: ReceiverStockTrade) {
        self.stock = stock
    }

    func execute() {
        stock.sell()
    }
}

// Receiver
class ReceiverStockTrade {
    func buy() {
        print("Buy stocks")
    }

    func sell() {
        print("Sell stocks")
    }
}

// Client Code
let trader = ReceiverStockTrade()
let buyStock = BuyStockOrder(stock: trader)
let sellStock = SellStockOrder(stock: trader)

let agent = Agent()
agent.placeOrder(command: buyStock)
agent.placeOrder(command: sellStock)

/*
Buy stocks
Sell stocks
*/

File: Design-Patterns-Using-CSharp/Command/CommandPatternStock.cs
using System;

// Agent (Invoker)
class Agent
{
    public void PlaceOrder(Order command)
    {
        command.Execute();
    }
}

// Order (Command)
abstract class Order
{
    public abstract void Execute();
}

// BuyStockOrder (ConcreteCommand)
class BuyStockOrder : Order
{
    private ReceiverStockTrade stock;

    public BuyStockOrder(ReceiverStockTrade stock)
    {
        this.stock = stock;
    }

    public override void Execute()
    {
        stock.Buy();
    }
}

// SellStockOrder (ConcreteCommand)
class SellStockOrder : Order
{
    private ReceiverStockTrade stock;

    public SellStockOrder(ReceiverStockTrade stock)
    {
        this.stock = stock;
    }

    public override void Execute()
    {
        stock.Sell();
    }
}

// Receiver
class ReceiverStockTrade
{
    public void Buy()
    {
        Console.WriteLine("Buy stocks");
    }

    public void Sell()
    {
        Console.WriteLine("Sell stocks");
    }
}

// Client code
class CommandPatternStock
{
    static void Main(string[] args)
    {
        ReceiverStockTrade trader = new ReceiverStockTrade();
        BuyStockOrder buyStock = new BuyStockOrder(trader);
        SellStockOrder sellStock = new SellStockOrder(trader);

        Agent agent = new Agent();
        agent.PlaceOrder(buyStock);
        agent.PlaceOrder(sellStock);
    }
}

/*
Buy stocks
Sell stocks
*/